pkg/gcp/ocpgwdeployer.go

```diff
@@ -44,11 +44,23 @@ type ocpGatewayDeployer struct {
 }
 
 // NewOcpGatewayDeployer returns a GatewayDeployer capable of deploying gateways using OCP.
-func NewOcpGatewayDeployer(info CloudInfo, msDeployer ocp.MachineSetDeployer, instanceType, image string,
+
+func NewOcpGatewayDeployer(info CloudInfo, //nolint: gocritic // Ignore 'hugeParam' - pass by value for CloudInfo is intentional.
+	msDeployer ocp.MachineSetDeployer, instanceType, image string,
 	k8sClient k8s.Interface,
 ) api.GatewayDeployer {
+	cloudInfo := info
+
+	if cloudInfo.VpcName == "" {
+		cloudInfo.VpcName = info.InfraID + "-network"
+	}
+
+	if cloudInfo.PublicSubnetName == "" {
+		cloudInfo.PublicSubnetName = info.InfraID + "-worker-subnet"
+	}
```

Comment:
No need to make another copy of 'info'.

```suggestion
	if info.VpcName == "" {
		info.VpcName = info.InfraID + "-network"
	}

	if info.PublicSubnetName == "" {
		info.PublicSubnetName = info.InfraID + "-worker-subnet"
	}
```

Reply:
Done.